src/libraries/System.Runtime/tests/System.Reflection.Tests/FieldInfoTests.cs

```diff
@@ -82,23 +149,75 @@ public void GetValue_Invalid(Type type, string name, object obj, Type exceptionT
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.s_stringField_Set), null, "new", "new" };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.s_objectField_Set), null, MyStruct.s_objectField, MyStruct.s_objectField };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.s_intPtr_Set), null, MyStruct.s_intPtrForComparison, MyStruct.s_intPtrForComparison };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.threadStatic_intField_Set), null, 100, 100 };
+
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.boolField), new MyStruct(), true, true };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.boolField), new MyStruct(), false, false };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.intField), new MyStruct(), 1002, 1002 };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.stringField), new MyStruct(), "new", "new" };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.objectField), new MyStruct(), MyStruct.s_objectField, MyStruct.s_objectField };
+            yield return new object[] { typeof(MyStruct), nameof(MyStruct.intPtr), new MyStruct(), MyStruct.s_intPtrForComparison, MyStruct.s_intPtrForComparison };
         }
 
         [Theory]
         [MemberData(nameof(SetValue_TestData))]
         public void SetValue(Type type, string name, object obj, object value, object expected)
+        {
+            FieldInfo fieldInfo = GetField(type, name);
+            object original = fieldInfo.GetValue(obj);
+            try
+            {
+                fieldInfo.SetValue(obj, value);
+                Assert.Equal(expected, fieldInfo.GetValue(obj));
+
+                // Perform a second time to rule out cases of slow-path vs. fast-path.
+                fieldInfo.SetValue(obj, value);
+                Assert.Equal(expected, fieldInfo.GetValue(obj));
+            }
+            finally
+            {
+                fieldInfo.SetValue(obj, original);
+            }
+        }
+
+        public static IEnumerable<object[]> SetValue_TestData_FunctionPointers()
+        {
+            yield return new object[] { typeof(MyStructWithFunctionPointers), nameof(MyStructWithFunctionPointers.s_fcnPtr_Set), null, (IntPtr)201, (IntPtr)201 };
+            yield return new object[] { typeof(MyStructWithFunctionPointers), nameof(MyStructWithFunctionPointers.fcnPtr), new MyStructWithFunctionPointers(), (IntPtr)200, (IntPtr)200 };
+        }
+
+        [Theory]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/97833", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
+        [MemberData(nameof(SetValue_TestData_FunctionPointers))]
+        public void SetValueWithFunctionPointers(Type type, string name, object obj, object value, object expected)
```

Comment:
NIT: Seems forgot the 2nd call: `// Perform a second time to rule out cases of slow-path vs. fast-path.`

Reply:
Sure. Thanks